storage/rocksdb/clone/common.h

```diff
@@ -0,0 +1,751 @@
+      case ER_NET_READ_INTERRUPTED:
+      case ER_NET_ERROR_ON_WRITE:
+      case ER_NET_WRITE_INTERRUPTED:
+      case ER_NET_WAIT_ERROR:
+        my_error(error_copy, MYF(0));
+        break;
+      case ER_CLONE_PROTOCOL:
+      case ER_INTERNAL_ERROR:
+        my_error(error_copy, MYF(0), "MyRocks clone error in another thread");
+        break;
+        // Errors with specific file paths
+      case ER_FILE_EXISTS_ERROR:
+      case ER_ERROR_ON_WRITE:
+      case ER_CANT_OPEN_FILE: {
+        assert(!error_path_copy.empty());
+        char errbuf[MYSYS_STRERROR_SIZE];
+        my_error(error_copy, MYF(0), error_path_copy.c_str(), errno_copy,
+                 my_strerror(errbuf, sizeof(errbuf), errno_copy));
+        break;
+      }
+      case ER_CLONE_DONOR:
+        /* Will get the error message from remote */
+        break;
+      default:
+        // TODO(laurynas): remove this assert at the end of development, until
+        // then, keep on adding more specific cases above as needed. InnoDB
+        // handles:
+        // - ER_CANT_CREATE_FILE
+        // - ER_ERROR_ON_READ
+        // - ER_WRONG_VALUE
+        assert(0);
+
+        my_error(ER_INTERNAL_ERROR, MYF(0),
+                 "MyRocks clone error in another thread");
+    }
+
+    return error_copy;
+  }
+
+  // The clone plugin always expects the master task ID to be zero, including
+  // the case of multiple threads using the same ID at the same time.
+  static constexpr auto master_task_id = 0;
+
+  session(const session &) = delete;
+  session(session &&) = delete;
+  session &operator=(const session &) = delete;
+  session &operator=(session &&) = delete;
+
+ protected:
+  session() {
+#ifdef HAVE_PSI_INTERFACE
+    mysql_rwlock_init(key_rwlock_clone_task_id_set, &task_ids_lock);
+    mysql_mutex_init(clone_master_task_remaining_mutex_key,
+                     &master_task_remaining_mutex, MY_MUTEX_INIT_FAST);
+    mysql_cond_init(rdb_signal_clone_master_task_remaining_key,
+                    &master_task_remaining_signal);
+    mysql_mutex_init(clone_error_mutex_key, &error_mutex, MY_MUTEX_INIT_FAST);
+#else
+    mysql_rwlock_init(nullptr, &task_ids_lock);
+    mysql_mutex_init(nullptr, &master_task_remaining_mutex, MY_MUTEX_INIT_FAST);
+    mysql_cond_init(nullptr, &master_task_remaining_signal);
+    mysql_mutex_init(nullptr, &error_mutex, MY_MUTEX_INIT_FAST);
+#endif
+  }
+
+  ~session() = default;
+
+ private:
+  void register_task_id(uint task_id) {
+    mysql_rwlock_wrlock(&task_ids_lock);
+    const auto insert_result [[maybe_unused]] = task_ids.emplace(task_id);
+    mysql_rwlock_unlock(&task_ids_lock);
+
+    assert(insert_result.second);
+  }
+
+  mutable mysql_rwlock_t task_ids_lock;
+  std::unordered_set<uint> task_ids;
+  static std::atomic<uint> next_task_id;
+
+  mutable mysql_cond_t master_task_remaining_signal;
+  mutable mysql_mutex_t master_task_remaining_mutex;
+
+  mutable mysql_mutex_t error_mutex;
+  int error = 0;
```

Comment:
In other rocksdb code, we try to denote class members with 'm_' prefix. Would it be too difficult to make these cosmetic changes? If it is, maybe consider `error_` or for this member since error is probably a common local variable name.

Reply:
Adding `m_` is not hard (C++ LSP servers support this kind of refactoring), so I'll do just that

Reply:
Pushed a commit, will squash later